addons/payment_mollie_official/models/payment_acquirer.py

```diff
@@ -0,0 +1,172 @@
+# -*- coding: utf-8 -*-
+from mollie.api.client import Client
+from odoo import models, fields, api, _
+from odoo.http import request
+
+import base64
+import requests
+import logging
+_logger = logging.getLogger(__name__)
+
+
+class PaymentAcquirer(models.Model):
+    _inherit = 'payment.acquirer'
+
+    _mollie_client = Client()
+
+    provider = fields.Selection(selection_add=[('mollie', 'Mollie')])
+    mollie_api_key_test = fields.Char(
+        'Mollie Test API key', size=40, required_if_provider='mollie',
+        groups='base.group_user')
+    mollie_api_key_prod = fields.Char('Mollie Live API key', size=40,
+                                      required_if_provider='mollie',
+                                      groups='base.group_user')
+    dashboard_url = fields.Char(string="Dashboard URL")
+    method_ids = fields.One2many('payment.acquirer.method',
+                                 'acquirer_id', 'Supported methods')
+
+    @api.model
+    def _get_main_mollie_provider(self):
+        return self.sudo().search([('provider', '=', 'mollie')], order="id",
+                                  limit=1) or False
+
+    @api.onchange('mollie_api_key_test')
+    def _onchange_mollie_api_key_test(self):
+        if self.mollie_api_key_test:
+            if not self.mollie_api_key_test[:5] == 'test_':
+                return {'warning':
+                        {'title': "Warning",
+                         'message': "Value of Test API Key is not valid."
+                         " Should begin with 'test_'", }}
+
+    @api.onchange('mollie_api_key_prod')
+    def _onchange_mollie_api_key_prod(self):
+        if self.mollie_api_key_prod:
+            if not self.mollie_api_key_prod[:5] == 'live_':
+                return {
+                    'warning': {'title': "Warning",
+                                'message': "Value of Live API Key is not "
+                                "valid. Should begin with 'live_'", }}
+
+    def _get_mollie_api_keys(self, state):
```

Comment:
not compatible with v13, it's either `test` or `enabled` (or `disabled`)

Reply:
also, why would you need to pass the state as a arg?
and shouldn't this be self.ensure_one()?